src/kili/mutations/asset/__init__.py

```diff
@@ -240,16 +260,87 @@ def generate_variables(batch: Dict) -> Dict:
         return [item for batch_list in formated_results for item in batch_list]
 
     @typechecked
-    def delete_many_from_dataset(self, asset_ids: List[str]) -> Asset:
+    def rename_assets(
+        self,
+        asset_ids: List[str],
+        new_external_ids: List[str],
+    ) -> List[Dict]:
+        """Update the names (external IDs) of one or more assets.
+
+        Args:
+            asset_ids: The asset IDs to modify.
+            new_external_ids: The new external IDs of the assets.
+
+        Returns:
+            A result object which indicates if the mutation was successful,
+                or an error message.
+
+        Examples:
+            >>> kili.rename_assets(
+                    asset_ids=["ckg22d81r0jrg0885unmuswj8", "ckg22d81s0jrh0885pdxfd03n"],
+                    new_external_ids=["asset1", "asset2"],
+                )
+        """
+
+        parameters = {
+            "asset_ids": asset_ids,
+            "new_external_ids": new_external_ids,
+            "json_metadatas": None,
+            "to_be_labeled_by_array": None,
+        }
+        properties_to_batch = process_update_properties_in_assets_parameters(parameters)
+
+        def generate_variables(batch: Dict) -> Dict:
+            data = {
+                "externalId": batch["new_external_ids"],
+                "jsonMetadata": batch["json_metadatas"],
+                "toBeLabeledBy": batch["to_be_labeled_by_array"],
+                "shouldResetToBeLabeledBy": batch["should_reset_to_be_labeled_by_array"],
+            }
+            data_array = [dict(zip(data, t)) for t in zip(*data.values())]
+            return {
+                "whereArray": [{"id": asset_id} for asset_id in batch["asset_ids"]],
+                "dataArray": data_array,
+            }
+
+        results = _mutate_from_paginated_call(
+            self,
+            properties_to_batch,
+            generate_variables,
+            GQL_UPDATE_PROPERTIES_IN_ASSETS,
+        )
+        formated_results = [format_result("data", result, Asset) for result in results]
+        return [item for batch_list in formated_results for item in batch_list]
+
+    @typechecked
+    def delete_many_from_dataset(
+        self,
+        asset_ids: Optional[List[str]] = None,
+        external_ids: Optional[List[str]] = None,
+        project_id: Optional[str] = None,
+    ) -> Asset:
         """Delete assets from a project.
 
         Args:
-            asset_ids: The list of identifiers of the assets to delete.
+            asset_ids: The list of asset internal IDs to delete.
+            external_ids: The list of asset external IDs to delete.
+            project_id: The project ID. Only required if `external_ids` argument is provided.
 
         Returns:
             A result object which indicates if the mutation was successful,
                 or an error message.
         """
+        if asset_ids is None and external_ids is None:
```

Comment:
you may add a helper to factorize this: `get_asset_ids_or_throw_error(asset_ids, external_ids, project_id)`